metadata-service/auth-impl/src/main/java/com/datahub/authorization/fieldresolverprovider/ContainerFieldResolverProvider.java

```diff
@@ -0,0 +1,80 @@
+package com.datahub.authorization.fieldresolverprovider;
+
+import static com.linkedin.metadata.Constants.*;
+
+import com.datahub.authorization.EntityFieldType;
+import com.datahub.authorization.EntitySpec;
+import com.datahub.authorization.FieldResolver;
+import com.linkedin.common.urn.Urn;
+import com.linkedin.common.urn.UrnUtils;
+import com.linkedin.container.Container;
+import com.linkedin.entity.EntityResponse;
+import com.linkedin.entity.client.SystemEntityClient;
+import io.datahubproject.metadata.context.OperationContext;
+import java.util.*;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+/** Provides field resolver for container given entitySpec */
+@Slf4j
+@RequiredArgsConstructor
+public class ContainerFieldResolverProvider implements EntityFieldResolverProvider {
```

Comment:
like domains, we should probably recursively check parent domains of domains right? in case you have permissions based on a parent domain of the domain on an entity. for example.